packages/styled/types/tests.tsx

```diff
@@ -217,3 +217,40 @@ const Input5 = styled.input`
   // $ExpectError
   ;<StyledCompWithoutAs as={Section} />
 }
+
+{
+  type StyledOptsParameterized = StyledOptions<{ a: string; b: number }>
+
+  const styledOpts0: StyledOptions = {
+    label: 'foo',
+    target: 'bar',
+    shouldForwardProp: p => true
+  }
+  const styledOpts1: StyledOptsParameterized = {
+    label: 'foo',
+    target: 'bar',
+    shouldForwardProp: (p: 'a' | string) => true
+  }
+  const styledOptsBroken0: StyledOptsParameterized = {
+    label: 'foo',
+    target: 'bar',
+    shouldForwardProp: (p: 'c' | number) => true // $ExpectError
+  }
+  const styledOptsBroken1: StyledOptions = {
+    label: 1, // $ExpectError
+    target: null, // $ExpectError
+    shouldForwardProp: (p1, p2) => true // $ExpectError
+  }
+
+  interface Props {
+    foo: boolean
+  }
+  const options: StyledOptions<JSX.IntrinsicElements['div']> = {
+    label: 'foo',
+    target: 'bar',
+    shouldForwardProp: p => true
+  }
+  const test1 = styled.div({ color: 'red' }, options)
+  const test2 = styled.div<Props>({ color: 'red' }, options)
+  const test3 = styled('div', options)({ color: 'red' })
```

Comment:
Could you also add a similar test but for a composite component (non-HTML)?

Reply:
Sure.